### PR TITLE
Infrastructure deployments report

### PR DESCRIPTION
**Pull request by poornima-krishnasamy**
Related to: https://github.com/ministryofjustice/cloud-platform/issues/3879

This PR creates a report of successful and failed deployments of cloud -platform-infrastructure repo for past 12 months

### CODE REVIEW

File: reports/infrastructure-deployments/main.go [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+	token          = flag.String("token", os.Getenv("GITHUB_OAUTH_TOKEN"), "Personal access token for GitHub API.")
+
+	endPoint  = *hoodawHost + *hoodawEndpoint
+	infraPath = "terraform/aws-accounts/cloud-platform-aws"
+
+	// Number of months to generate report
+	numMonths = 12
+	// number of PRs to fetch per month. set to 100 with assumption
+	// that there are no more than 100 prs in the infrastructure repo
+	prCount = 100
```
Comment (jasonBirchall) [inline reports/infrastructure-deployments/main.go:42]: I don't see `endpoint, infrapath, numMonths and prCount` as variables. These are static values and should be constants.

File: reports/infrastructure-deployments/main.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+}
+
+func main() {
+	flag.Parse()
+	// Set nthMonth to count 0 which is the current month.
+	// This report generated data for past 12 months based on current month
+	nthMonth := 0
+	infraReport := make([]map[string]string, 0)
+
+	for nthMonth < numMonths {
```
Comment (jasonBirchall) [inline reports/infrastructure-deployments/main.go:69]: For readability can we `for m:=0; m < numMonths; m++ {`?

File: reports/infrastructure-deployments/main_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+func Test_getFirstLastDayofMonth(t *testing.T) {
+	type args struct {
+		nthMonth int
+	}
+	tests := []struct {
+		name string
+		args args
+		want date
+	}{
+		// TODO: Add test cases.
```
Comment (jasonBirchall) [inline reports/infrastructure-deployments/main_test.go:17]: Do we mean to test here?

**Review by jasonBirchall — CHANGES_REQUESTED** [review]
Just a few small changes

File: reports/infrastructure-deployments/main.go (hunk 1 above)
Comment (poornima-krishnasamy) [reply reports/infrastructure-deployments/main.go]: Thanks @jasonBirchall . Updated. The endpoint is from 2 vars so left it as a var

File: reports/infrastructure-deployments/main.go (hunk 2 above)
Comment (poornima-krishnasamy) [reply reports/infrastructure-deployments/main.go]: Updated

File: reports/infrastructure-deployments/main_test.go (hunk 3 above)
Comment (poornima-krishnasamy) [reply reports/infrastructure-deployments/main_test.go]: Yes, but they are not yet ready. I have removed until i have a working test

**Review by jasonBirchall — APPROVED** [review]

**Pull request merged**